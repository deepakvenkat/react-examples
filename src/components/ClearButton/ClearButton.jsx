import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';

import CalculatorButton from '../CalculatorButton';

const styles = {
  root: {
    width: '225px',
    backgroundColor: 'white',
    '& .MuiButton-label': {
      textTransform: 'lowercase',
    }
  },
}

const ClearButton = ({
  classes,
}) => (
  <CalculatorButton
    type="operator"
    text="clear"
    classes={classes}
  />
);

ClearButton.propTypes = {
  classes: PropTypes.shape({}).isRequired,
};

export default withStyles(styles)(ClearButton);
