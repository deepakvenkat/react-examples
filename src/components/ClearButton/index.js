import ClearButton from './ClearButton.jsx';

export default ClearButton;
