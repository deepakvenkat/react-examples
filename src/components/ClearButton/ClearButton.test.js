import React from 'react';
import { render } from '@testing-library/react';
import ClearButton from './ClearButton';

test('renders the clearbutton', () => {
  const { asFragment } = render(
    <ClearButton />
  );
  expect(asFragment()).toMatchSnapshot();
});
