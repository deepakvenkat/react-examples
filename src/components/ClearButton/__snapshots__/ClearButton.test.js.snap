// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the clearbutton 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained CalculatorButton-root-2 CalculatorButton-root-3 ClearButton-root-1 MuiButton-disableElevation"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      clear
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
